pkg/systemd/services/service-details.jsx

```diff
@@ -308,6 +324,22 @@ export class ServiceDetails extends React.Component {
         }
     }
 
+    pinUnit() {
+        let pinnedUnits = JSON.parse(localStorage.getItem('pinnedUnits')) || [];
+
+        this.setState({ waitsAction: true });
+
+        if (this.props.unit.IsPinned) {
+            pinnedUnits = pinnedUnits.filter(unitId => unitId != this.props.unit.path);
+        } else {
+            pinnedUnits.push(this.props.unit.path);
+        }
+
+        localStorage.setItem('pinnedUnits', JSON.stringify(pinnedUnits));
+        dispatchEvent(new Event('storage'));
+        this.setState({ waitsAction: false });
+    }
```

Comment:
We set `waitsAction: true` only in cases when we don't want user to do any other action with the service, until the previous actions is finished, otherwise those 2 actions could trigger 2 conflicting API calls. I think setting `waitsAction` here is unecessary because:
1. It makes sense to set it only with asynchronous methods, which might take a long time. localStorage.setItem is however a synchronous method, so it resolves immediatelly and there is no time for user to do another action which would trigger some other API call.
2. Even if localStorage.setItem was an asynchronous method, it doesn't call systemd, it doesn't change configuration of the unit file, so there is no way `localStorage.setItem` would be conflicting with some other action/API call.

Reply:
So I think this code can be like this:
```suggestion
        if (this.props.unit.IsPinned) {
            pinnedUnits = pinnedUnits.filter(unitId => unitId != this.props.unit.path);
        } else {
            pinnedUnits.push(this.props.unit.path);
        }

        localStorage.setItem('pinnedUnits', JSON.stringify(pinnedUnits));
        dispatchEvent(new Event('storage'));
    }
```

Reply:
I see, thank you for explaining when `waitsAction` should be used!